client/src/components/growth-book/growth-book-wrapper.tsx

```diff
@@ -63,12 +55,32 @@ interface UserAttributes {
   signedIn?: true;
 }
 
+const mapDispatchToProps = (dispatch: Dispatch) =>
+  bindActionCreators({ executeGA }, dispatch);
+
 const GrowthBookWrapper = ({
   children,
   isSignedIn,
   user,
-  userFetchState
+  userFetchState,
+  executeGA
 }: GrowthBookWrapper) => {
+  const growthbook = useMemo(
+    () =>
+      new GrowthBook({
+        trackingCallback: (experiment, result) => {
+          executeGA({
+            event: 'experiment_viewed',
+            event_category: 'experiment',
+            experiment_id: experiment.key,
+            variation_id: result.variationId
+          });
+        }
+      }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
```

Comment:
What's the rationale for moving this from being a global? It seems like memo does the same thing, but in a more complicated way.

Reply:
I was just trying to use executeGA which is GrowthBookWrapper prop, so I moved the growthbook variable initialization inside the component to use the prop.

Reply:
@ojeytonwilliams, let me know if you require further changes.

Reply:
Maybe not this PR, but there doesn't seem to be any reason to involve redux. We dispatch `executeGA` purely to trigger the saga. Given that, we could just call `TagManager.dataLayer` directly and remove the `executeGA` action.

Reply:
That makes sense. The function was directly pushing to dataLayer removing the TagManager completely. I just wanted all GA events to be created similarly to avoid any unknown abstraction issues.